dart/constraint/ConstrainedGroup.h

```diff
@@ -107,7 +108,7 @@ class ConstrainedGroup
   std::vector<ConstraintBase*> mConstraints;
 
   ///
-  dynamics::Skeleton* mRootSkeleton;
+  std::shared_ptr<dynamics::Skeleton> mRootSkeleton;
```

Comment:
If we want to get really fancy, one option might be to abstract out the reference counting functionality of BodyNode that is needed for BodyNodePtr. If the referencing counting is put into a base class that gets inherited by BodyNodePtr, then maybe BodyNodePtr can just depend on the very simple base class instead of depending on a fully defined BodyNode class. Then we can safely put BodyNodePtr and JointPtr into DynamicsTypes.h.

I don't know for sure if this would work, but I think it's doable with some clever templating tricks.